File: shell_sort.py
def shell_sort(a):
    n = len(a)
    gap = n//2
    while gap:
        for i in range(gap,n):
            j = i
            temp = a[i]
            while j>gap-1 and a[j-gap]>temp:
                a[j] = a[j-gap]
                j -= gap
            a[j] = temp
        gap = gap // 2

File: javascript和css学习/shell_sort.js
function shellSort(a){
    var n = a.length;
    var gap = Math.floor(n/2);
    while(gap){
        for(var i=gap;i<n;i+=gap){
            var j=i;
            var temp = a[i];
            while(j>0 && temp<a[j-gap]){
                a[j] = a[j-gap];
                j -= gap;
            }
            a[j]=temp;
        }
        gap = Math.floor(gap/2);
    }
}